OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+        mutable.Map.empty[(ObjectType, String), ListBuffer[(Method, String, Int)]] //Class and name of the Method that returns the searched intent -> the intent sending methods and the component type
+    val returningIntentsMap: mutable.Map[(ObjectType, String), ListBuffer[ImplicitIntent]] =
+        mutable.Map.empty[(ObjectType, String), ListBuffer[ImplicitIntent]] //class and method that returns an intent -> intent
+
+    var leftOverMethods: Set[Method] = Set.empty[Method]
+    var notStarted = true
+
+    val source: BufferedSource = Source.fromInputStream(this.getClass.getResourceAsStream("AndroidCallbackList.txt"))
+    val callbackList: Set[ObjectType] = source.getLines.map(ObjectType(_)).toSet
+    source.close
+
+    /**
+     * Calls all necessary methods to generate the call graph edges that are Android specific.
+     *
+     * @param manifest AndroidManifest.xml
+     */
+    def performAnalysis(
+        project: SomeProject
+    ): ProperPropertyComputationResult = {
+        val manifestParsingResultOption = project.get(AndroidManifestKey)
+        if (manifestParsingResultOption.isDefined) {
+            val manifestParsingResult = manifestParsingResultOption.get
+            intentFilters ++= manifestParsingResult._2
+            generateLifecycleCallbacks(manifestParsingResult._1)
+        }
+        searchIntentsAndCallbacks()
+    }
+
+    /**
+     * Starts the second phase of the analysis, after all methods are processed once to collect information on intents
+     * and intent filters.
+     *
+     * @param project The analysed project.
+     * @return Returns the calls originating from inter-component-communication.
+     */
+    def startSecondPhase(project: SomeProject): ProperPropertyComputationResult = {
+        intentMatching()
+
+        val decMeths = project.get(DeclaredMethodsKey)
+        val contexts = project.get(SimpleContextsKey)
+        val resList = ListBuffer.empty[PartialResult[_, _ >: Null <: Property]]
+        edges.foreach { e ⇒
+            val calls = new IndirectCalls()
+            calls.addCall(contexts(decMeths(e._1._1)), e._1._2, contexts(decMeths(e._2)))
+            resList ++= calls.partialResults(contexts(decMeths(e._1._1))).seq
+        }
+
+        Results(resList)
+    }
+
+    /**
+     * Generates the lifecycle callbacks for all android components of the project.
+     * Uses the results from AndroidManifestKey.
+     */
+    def generateLifecycleCallbacks(componentMap: Map[String, ListBuffer[ClassFile]]): Unit = {
+        var fullLifecycle = true //cg is only sound if this is true
+        var aLifecycleList = List.empty[Method]
+        var sLifecycleList = List.empty[Method]
+        //find default super methods
+        if (fullLifecycle) {
+            val activityCF = project.classFile(androidActivityOT).orNull
+            if (activityCF != null) {
+                val aOnCreate = activityCF.findMethod(onCreate).head
```

Comment:
Are you sure that the first method (`head`) is the right thing to use here? What if there are multiple methods of that name? Maybe they all have to be analyzed? Or does this apply only to a specific method with a known descriptor? In that case, also use the descriptor to ensure that you only find that one method.